lib/kubernetes-deploy/runner.rb

```diff
@@ -295,25 +296,38 @@ def validate_configuration
       KubernetesDeploy.logger.info("All required parameters and files are present")
     end
 
-    def update_tprs(resources)
-      resources.each do |r|
+    def deploy_resources(resources, prune: false)
+      KubernetesDeploy.logger.info("Deploying resources:")
+
+      # Apply can be done in one large batch, the rest have to be done individually
+      applyables, individuals = resources.partition { |r| r.deploy_method == :apply }
+
+      individuals.each do |r|
         KubernetesDeploy.logger.info("- #{r.id}")
         r.deploy_started = Time.now.utc
-        _, _, st = run_kubectl("replace", "-f", r.file.path)
+        case r.deploy_method
+        when :replace
+          _, _, st = run_kubectl("replace", "-f", r.file.path)
+        when :replace_force
+          _, _, st = run_kubectl("replace", "--force", "-f", r.file.path)
+        end
 
         unless st.success?
           # it doesn't exist so we can't replace it
-          run_kubectl("create", "-f", r.file.path)
+          _, err, st = run_kubectl("create", "-f", r.file.path)
+          unless st.success?
+            raise FatalDeploymentError, <<~MSG
```

Comment:
This gem is for use in Shipit, which afaik we're keeping 2.2 compatible, so I've been avoiding 2.3-specific features, as nice as they are. You have `.strip_heredoc` from ActiveSupport available though.